pkg/lwip/contrib/netdev/lwip_netdev.c

```diff
@@ -96,83 +96,82 @@ err_t lwip_netdev_init(struct netif *netif)
     netif->hostname = "riot";
 #endif /* LWIP_NETIF_HOSTNAME */
 
-    /* XXX: for now assume its Ethernet, since netdev is implemented only by ethernet drivers */
```

Comment:
Note: This comment is obsolete, below both Ethernet and IEEE 802.15.4 devices are handled differently. There is no "this must be Ethernet" assumption baked in anywhere.